builtin/providers/oneandone/provider.go

```diff
@@ -0,0 +1,94 @@
+package oneandone
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"github.com/1and1/oneandone-cloudserver-sdk-go"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+	"golang.org/x/crypto/ssh"
+	"io/ioutil"
+)
+
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"token": {
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ONEANDONE_TOKEN", nil),
+				Description: "1&1 token for API operations.",
+			},
+			"retries": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  50,
+			},
+			"endpoint": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  oneandone.BaseUrl,
+			},
+		},
+		ResourcesMap: map[string]*schema.Resource{
+			"oneandone_server":            resourceOneandOneServer(),
+			"oneandone_firewall_policy":   resourceOneandOneFirewallPolicy(),
+			"oneandone_private_network":   resourceOneandOnePrivateNetwork(),
+			"oneandone_public_ip":         resourceOneandOnePublicIp(),
+			"oneandone_shared_storage":    resourceOneandOneSharedStorage(),
+			"oneandone_monitoring_policy": resourceOneandOneMonitoringPolicy(),
+			"oneandone_loadbalancer":      resourceOneandOneLoadbalancer(),
+			"oneandone_vpn":               resourceOneandOneVPN(),
+		},
+		ConfigureFunc: providerConfigure,
+	}
+}
+
+func providerConfigure(d *schema.ResourceData) (interface{}, error) {
+	var endpoint string
+	if d.Get("endpoint").(string) != oneandone.BaseUrl {
+		endpoint = d.Get("endpoint").(string)
+	}
+	config := Config{
+		Token:    d.Get("token").(string),
+		Retries:  d.Get("retries").(int),
```

Comment:
"retries" are not used in `provider.go` but are used pretty much in ever 1and1 resource.